src/System.Text.Json/src/System/Text/Json/Node/JsonNumber.cs

```diff
@@ -0,0 +1,614 @@
+        /// </exception>
+        public byte GetByte() => byte.Parse(_value);
+
+        /// <summary>
+        ///   Converts the numeric value of this instance to its <see cref="short"/> equivalent.
+        /// </summary>
+        /// <returns>A <see cref="short"/> equivalent to the number stored by this instance.</returns>
+        /// <exception cref="OverflowException">
+        ///   <see cref="JsonNumber"/> represents a number less than <see cref="short.MinValue"/> or greater than <see cref="short.MaxValue"/>.
+        /// </exception>
+        /// <exception cref="FormatException">
+        ///   <see cref="JsonNumber"/> represents a number in a format not compliant with <see cref="short"/>.
+        /// </exception>
+        public short GetInt16() => short.Parse(_value);
+
+        /// <summary>
+        ///   Converts the numeric value of this instance to its <see cref="int"/> equivalent.
+        /// </summary>
+        /// <returns>A <see cref="int"/> equivalent to the number stored by this instance.</returns>
+        /// <exception cref="OverflowException">
+        ///   <see cref="JsonNumber"/> represents a number less than <see cref="int.MinValue"/> or greater than <see cref="int.MaxValue"/>.
+        /// </exception>
+        /// <exception cref="FormatException">
+        ///   <see cref="JsonNumber"/> represents a number in a format not compliant with <see cref="int"/>.
+        /// </exception>
+        public int GetInt32() => int.Parse(_value);
+
+        /// <summary>
+        ///   Converts the numeric value of this instance to its <see cref="long"/> equivalent.
+        /// </summary>
+        /// <returns>A <see cref="long"/> equivalent to the number stored by this instance.</returns>
+        /// <exception cref="OverflowException">
+        ///   <see cref="JsonNumber"/> represents a number less than <see cref="long.MinValue"/> or greater than <see cref="long.MaxValue"/>.
+        /// </exception>
+        /// <exception cref="FormatException">
+        ///   <see cref="JsonNumber"/> represents a number in a format not compliant with <see cref="long"/>.
+        /// </exception>
+        public long GetInt64() => long.Parse(_value);
+
+        /// <summary>
+        ///   Converts the numeric value of this instance to its <see cref="float"/> equivalent.
+        /// </summary>
+        /// <returns>A <see cref="float"/> equivalent to the number stored by this instance.</returns>
+        /// <exception cref="OverflowException">
+        ///   <see cref="JsonNumber"/> represents a number less than <see cref="float.MinValue"/> or greater than <see cref="float.MaxValue"/>.
+        /// </exception>
+        /// <exception cref="FormatException">
+        ///   <see cref="JsonNumber"/> represents a number in a format not compliant with <see cref="float"/>.
+        /// </exception>
+        public float GetSingle() => float.Parse(_value);
```

Comment:
For scientific mode values this could be

```C#
public float GetSingle()
{
    if (float.TryParse(_value, out float value))
    {
        return value;
    }

    return float.Parse(_value, NumberStyles.AllowExponent);
}
```

(and similarly for double)

Reply:
The default float parser already supports `NumberStyles.AllowExponent` (and a lot more). In fact, specifying `NumberStyles.AllowExponent` limites what float.Parse accepts (it would fail to parse decimals for instance).

https://github.com/dotnet/corefx/blob/cc5653fcceefce5eac38dc7847b12b371027f78f/src/Common/src/CoreLib/System/Globalization/NumberStyles.cs#L61-L62
https://github.com/dotnet/corefx/blob/cc5653fcceefce5eac38dc7847b12b371027f78f/src/Common/src/CoreLib/System/Single.cs#L298-L302

Same with double.

https://github.com/dotnet/corefx/blob/cc5653fcceefce5eac38dc7847b12b371027f78f/src/Common/src/CoreLib/System/Double.cs#L299-L303